app/src/main/java/com/cliqz/browser/inproductmessaging/MessagingHandler.kt

```diff
@@ -0,0 +1,59 @@
+package com.cliqz.browser.inproductmessaging
+
+import org.json.JSONArray
+import org.json.JSONException
+import java.util.*
+import timber.log.Timber;
+
+/**
+ * @author Ravjit Uppal
+ */
+class MessagingHandler private constructor() : MessageReceivedListener {
+
+    private var mMessages: MutableList<Message>? = null
```

Comment:
Let's have a non-null type and also not have one more field (`messages)` to access it.

```
lateinit var mMessages: MutableList<Message>
    private set
```

It is not anymore `Nullable` (with `?`), and can be accessed below as
`mMessages.size` instead of `mMessages!!.size`